runtime/common/src/crowdloan.rs

```diff
@@ -601,6 +594,29 @@ impl<T: Config> Module<T> {
 	pub fn crowdloan_kill(index: TrieIndex) -> child::KillChildStorageResult {
 		child::kill_storage(&Self::id_from_index(index), Some(T::RemoveKeysLimit::get()))
 	}
+
+	/// This function checks all conditions which would qualify a crowdloan has ended.
+	/// * If we have reached the `fund.end` block
+	///    * or
+	/// * If the first lease period the fund is trying to bid for has started already
+	///    * and
+	/// * If the fund has enough free funds to refund full raised amount.
```

Comment:
The docs are ambiguous: is the condition like `(fund.end || lease_period_started) && enough_free_funds` or like `fund.end || (lease_period_started && enough_free_funds)`?

Reply:
The former. I will try to improve